fvm/evm/emulator/state/code.go

```diff
@@ -0,0 +1,62 @@
+package state
+
+import (
+	"encoding/binary"
+	"fmt"
+)
+
+// CodeContainer contains codes and keeps
+// track of reference counts
+type CodeContainer struct {
+	code     []byte
+	refCount uint64
+}
+
+// NewCodeContainer constructs a new code container
+func NewCodeContainer(code []byte) *CodeContainer {
+	return &CodeContainer{
+		code:     code,
+		refCount: 1,
+	}
+}
+
+// CodeContainerFromEncoded constructs a code container from the encoded data
+func CodeContainerFromEncoded(encoded []byte) (*CodeContainer, error) {
+	if len(encoded) < 8 {
+		return nil, fmt.Errorf("invalid length for the encoded code container")
+	}
+	return &CodeContainer{
+		refCount: binary.BigEndian.Uint64(encoded[:8]),
+		code:     encoded[8:],
+	}, nil
```

Comment:
See comment above to keep reference to `encoded` buffer for reuse.

```suggestion
	return &CodeContainer{
		refCount: binary.BigEndian.Uint64(encoded[:8]),
		code:     encoded[8:],
		buffer:    encoded,
	}, nil
```